ui/app/registration/controllers/patientCommonController.js

```diff
@@ -16,9 +16,9 @@ angular.module('bahmni.registration')
             $scope.dobMandatory = appService.getAppDescriptor().getConfigValue("dobMandatory") || false;
             $scope.readOnlyExtraIdentifiers = appService.getAppDescriptor().getConfigValue("readOnlyExtraIdentifiers");
             $scope.showSaveConfirmDialogConfig = appService.getAppDescriptor().getConfigValue("showSaveConfirmDialog");
-            $scope.shouldShowSaveAndContinueButton = false;
+            $scope.showSaveAndContinueButton = false;
 
-            var savedButtonClicked = false;
+            var dontSaveButtonClicked = false;
```

Comment:
Change it to `cancelButtonClicked`, that's **a lot** clearer in the realm of OK/Cancel dialogs...